packages/swap/src/helpers/orders/getTotalFees.ts

```diff
@@ -1,29 +1,62 @@
-import {Balance, Numbers} from '@yoroi/types'
+import {Balance} from '@yoroi/types'
 import {Quantities} from '../../utils/quantities'
+import {ceilDivision} from '../../utils/ceilDivision'
+import {asQuantity} from '../../utils/asQuantity'
 
-/**
- * Calculate the total fees by summing the batcher fee and provider fee.
- * Later we can add the fronted feee
- *
- * @param {Balance.Quantity} batcherFee - The batcher fee as a balance quantity.
- * @param {Balance.Quantity} providerFee - The provider fee as a balance quantity.
- * @param {number} decimals - The number of decimal places for formatting the result.
- * @param {NumberLocale} numberLocale - The locale information for formatting numbers.
- *
- * @returns {string} The total fees as a string with specified decimals.
- */
-
-export const getTotalFees = (
-  batcherFee: Balance.Quantity,
-  providerFee: Balance.Quantity,
-  decimals: number,
-  numberLocale: Numbers.Locale,
-): string => {
-  const result = Quantities.denominated(
-    Quantities.sum([batcherFee, providerFee]),
-    decimals,
-  )
-
-  const [quantities] = Quantities.parseFromText(result, decimals, numberLocale)
-  return quantities
+export const getLiquidityProviderFee = (
+  poolFee: string,
+  sell: Balance.Amount,
+  sellTokenDecimals: number,
+): [string, Balance.Quantity] => {
+  const providerFee = Quantities.isZero(sell.quantity)
+    ? Quantities.zero
+    : asQuantity(
+        ceilDivision(
+          BigInt(Math.floor(Number(poolFee) * 1000)) * BigInt(sell.quantity),
+          100n * 1000n,
+        ).toString(),
+      )
+
+  return [Quantities.format(providerFee, sellTokenDecimals), providerFee]
+}
+
+export const getBatcherFee = (
+  batcherFee: Balance.Amount,
+  adaDecimals: number,
+): [string, Balance.Quantity] => {
+  if (batcherFee.tokenId !== '')
+    throw new Error('Unexpected batcher fee tokenId')
+
+  return [
+    Quantities.format(batcherFee.quantity, adaDecimals),
+    batcherFee.quantity,
+  ]
+}
+
+export const getMuesliswapFrontendFee = (
+  adaPrice: string,
+  milkInWallet: Balance.Quantity,
+  sell: Balance.Amount,
```

Comment:
@jorbuedo  not sure what `milkInWallet` means or maybe the name is a bit weird, and where this function `getMuesliswapFrontendFee` is called in the app. 🤔

Reply:
@SorinC6  milkInWallet would be how many milk tokens there are on the user wallet
It's part of the fee calculation for muesliswap frontend: https://docs.muesliswap.com/cardano/fee-structure

`getMuesliSwapFrontendFee` is not called anywhere yet. Discussed at https://emurgo.slack.com/archives/C02M83F3G72/p1695823475698869?thread_ts=1695822921.382099&cid=C02M83F3G72